apps/dashr-object-detection/app.R

```diff
@@ -0,0 +1,870 @@
+appName <- Sys.getenv("DASH_APP_NAME")
+pathPrefix <- sprintf("/%s/", appName)
+Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix, 
+          DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+
+setwd("app/apps/dashr-object-detection")
+
+
+library(dashR)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(dashPlayer)
+library(plotly)
+library(rlist)
+library(stringr)
+library(data.table)
+library(glue)
+library(compiler)
+library(magrittr)
+
+
+DEBUG <- T
+FRAMERATE <- 24.0
+f_read <- cmpfun(fread)
+data_table = cmpfun(data.table)
+
+gum <- cmpfun(glue)
+
+dict = cmpfun(list)
+
+aplly <- cmpfun(apply)
+laplly <- cmpfun(lapply)
+
+
+load.data <- cmpfun(function(path){
+  #  # Load data about a specific footage (given by the path). It returns a dictionary of useful variables such as
+  #  #  the dataframe containing all the detection and bounds localization, the number of classes inside that footage,
+  #  #  the matrix of all the classes in string, the given class with padding, and the root of the number of classes,
+  #  #  rounded.
+  # 
+  
+  # Load DT containing all the processed object detections inside the video
+  
+  Info.DT <- f_read(path)
+  
+  #cls.vec <- the dict of detected object classes for the .csv file
+  
+  cls.vec <- Info.DT[, class_str]
+  n.cls <- length(cls.vec)
+  title.vec <- unique(cls.vec)
+  
+  # Gets the smallest value needed to add to the end of the classes dict to get a square matrix
+  #   # n := argmin_{n : int} (n \geq n.c) \land (\sqrt{n} : int)
+  #   # to find n we do
+  
+  r.rnd <- ceiling(sqrt(n.cls))
+  
+  pad.ent<- as.integer(r.rnd^2 - n.cls)
+  pad.seq <- numeric(pad.ent)
+  #Pad the class.vec
+  cls.wpad <- c(cls.vec, pad.seq)
+  cls.mtx <- aplly(matrix(cls.wpad, nrow=r.rnd, ncol=r.rnd, byrow=T),2,rev)
+  
+  
+  
+  dat_dict = dict(INFO_DT = Info.DT, N_CLS=n.cls, MTX_CLS = cls.mtx, WPAD_CLS = cls.wpad, RT_RND = r.rnd, TITL=title.vec)
+  
+  
+  if(DEBUG)
+  {
+    print(gum('{path} loaded.'))
+  }
+  
+  
+  return(dat_dict)
+  
+})
+
+LABELS <- function(footage){
+  titles = dat.dict[[footage]]$TITL
+  l <- length(titles)
+  m <- ceiling(sqrt(l))
+  titles_pad <- c(titles, rep("", times=(m^2-l)))
+  
+  hover_text = titles_pad
+  
+  hover_text = matrix(hover_text, ncol = m, byrow=F)
+  
+  colnames(hover_text) <- 1:m
+  rownames(hover_text) <- 1:m
+  df <- data.table(hover_text, keep.rownames = T)
+  return(dashDataTable(
+    id = 'table',
+    columns = lapply(colnames(df), function(x) {
+      list(name = x, id = x)
+    }),
+    data = setNames(lapply(split(df, seq(nrow(df))), FUN = function (x) {as.list(x)}), NULL)
+  ))
+}
+
+
+#dict of optimized function through cmpfun
+load_data <- cmpfun(load.data)
+
+
+
+footage_labels <- c("james_bond", "zebra", "car_show_drone", "car_footage", "DroneCanalFestival","DroneCarFestival2","FarmDrone", "ManCCTV", "RestaurantHoldup")
+
+dat.dict <- dict( james_bond = load_data("data/james_bond_object_data.csv"), 
```

Comment:
Did you try caching this data? possibly using R.Cache and load / saveCache respectively inside the load.data function, seems you are loading it everytime. Fread and datatable uitlization is optimal though!